src/MLJTreeParzen.jl

```diff
@@ -262,10 +261,17 @@ function MLJTuning.models(
     end
 
     newstate = (space=space, trialhist=trialhist)
-    vector_of_metamodels = [
-        (modeltype(; candidate.hyperparams...), candidate)
-        for candidate in candidates
-    ]
+    vector_of_metamodels = Tuple{Any, Trials.Trial}[]
+
+    for candidate in candidates
+        metamodel = deepcopy(model)
+        for (hyperparam, value) in candidate.hyperparams
```

Comment:
How about "candidate.hyperparams" is an array of dictionary or a tuple of dictionary?

Reply:
`candidate.hyperparams` is a `Dict{Symbol, Any}()`, e.g. `Dict{Symbol, Any}(:gamma => 42.0, :max_depth => 11.0}`

Reply:
Oh sorry I think I missed your point which I think it's a valid one. Did you mean cases when space was defined differently which would then result in `candidate.hyperparams` to be other type? Such as for example there are other valid space definitions like this: `space = HP.Choice(:some_param, [1, 2])` or this `space = [HP.Choice(:param1, [10, 100]), HP.Uniform(:param2, 5., 10.)]`

Reply:
While arrays and tuples are valid space definitions in TreeParzen, the existing implementation of `MLJTreeParzenSpace`  space is defined as `Dict{Symbol}` only https://github.com/IQVIA-ML/TreeParzen.jl/blob/master/src/MLJTreeParzen.jl#L82 so a list or tuple won't go through. However, there is a possibility of layered/nested dictionary entries in the form of something like this: space = `Dict(:x => HP.Choice(:x, [spaceA, spaceB]))` where `spaceA` and `spaceB` are dictionaries so yes in that sense we'd pass an array of dictionaries. In the existing implementation this nesting is possible if a model container and its constructor are created as it won't just work by instantiating a model. So we'd need to look at keeping this nesting/layering possible after this change to mutate parameters.

Reply:
I will update documentation as creating a model container like described here: https://github.com/IQVIA-ML/TreeParzen.jl/blob/master/docs/examples/simple_mlj_demo/simple_mlj_demo.md#optimising-parameters-conditionally is no longer required